src/components/dashboard/dashboard.component.spec.ts

```diff
@@ -1,19 +1,30 @@
 import { ComponentFixture, TestBed } from '@angular/core/testing';
 import { DashboardComponent } from './dashboard.component';
-import { PreferenceModule } from '../../services/preference/preference.module';
+import { DataService } from '../../services/data/data.service';
+import { PreferenceService } from '../../services/preference/preference.service';
+import { PreferenceGroupModule } from '../preference-group/preference-group.module';
 
 describe('DashboardComponent', () => {
   let fixture: ComponentFixture<DashboardComponent>;
   let component: DashboardComponent;
+  let preferenceService: PreferenceService;
+  let dataService: DataService;
 
   beforeEach(() => {
+    preferenceService = new PreferenceService();
+    dataService = new DataService(preferenceService);
```

Comment:
I think we can assume these services work as expected when testing components.

Reply:
In general we try to mock almost all dependencies in order to not introduce potentially-breaking dependencies into unit tests, which ideally test only the single component (i.e. Dashboard) and do not need to care about changes made to PreferenceService or DataService. It also helps make the test a better form of documentation - i.e. I can clearly see from a test what I expect PreferenceService or DataService to provide to the Dashboard.

Perhaps this is different however - I have not looked too carefully at the dashboard code to see how / if these services are used. Feel free to ignore if I'm missing stuff.

Reply:
I agree it would be ideal. Do the mocked services extend the actual services and override methods to make them do nothing? Also, if we end up mocking every dependency, should we still import the module (as we discussed [here](https://github.com/googleinterns/guide-doge/pull/46#discussion_r452401076))?

Reply:
Yep, I think you would still import the module and then provide the mocked services in the `providers` array like you already are but the "service" would be something like:
```
let preferenceService: jasmine.SpyObj<PreferenceService>;

beforeEach(() => { 
  preferenceService = new jasmine.createSpyObj('preferenceService', ['data$, 'audification$', ...etc.]);

 ...

  providers: [
    { provide: PreferenceService, useValue: preferenceService,}
  ]
  ...
```

It is a bit of extra code and kind of annoying to set up, but does allow for more isolated unit tests.

Reply:
Ahh, didn't know you can create an entire spy object like that. That would be very useful. We will address it in #54, thanks!